Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector_test.py

```diff
@@ -0,0 +1,443 @@
+from ArmisEventCollector import Client, datetime, DemistoException, arg_to_datetime, EVENT_TYPE, EVENT_TYPES
+import pytest
+
+
+@pytest.fixture
+def dummy_client(mocker):
+    """
+    A dummy client fixture for testing.
+    """
+    mocker.patch.object(Client, 'is_valid_access_token', return_value=True)
+    return Client(base_url='test_base_url', api_key='test_api_key', access_token='test_access_token', verify=False, proxy=False)
+
+
+class TestClientFunctions:
+
+    def test_fetch_by_aql_query(self, mocker, dummy_client):
+        """
+        Given:
+            - A valid HTTP request parameters.
+        When:
+            - Fetching events.
+        Then:
+            - Send the HTTP request with valid parameters.
```

Comment:
Not sure that is what you checked cause you don't check what the mocked call was called with, maybe change it to validate basic http flow?

Reply:
This test checks the http request along with the pagination logic. I agree the docstring is not that accurate, changed it to better match the UT's purpose.